### PR TITLE
[FLOC-3848] Gce documentation

### PR DESCRIPTION
**Pull request by sarum90**
WIP, do not submit.

First draft of GCE documentation.

### CODE REVIEW

File: docs/spelling_wordlist.txt [hunk 1]
```diff
@@ -11,10 +11,11 @@ bugfix
 Buildbot
 CentOS
 centric
+checkbox
```
Comment (sarum90) [inline docs/spelling_wordlist.txt]: This is new. Everything else was the product of running `sort -f`

Comment (davidoliverSP2) [inline docs/spelling_wordlist.txt]: :+1:

File: docs/flocker-features/gce-configuration.rst [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+sufficient `scope
+<https://cloud.google.com/compute/docs/authentication#using>`_ to perform GCE
+operations.
+
+The required scope is ``https://www.googleapis.com/auth/compute``.
+In the GCE console this corresponds to the ``Allow API access to all Google Cloud services in the same project`` checkbox when you start an instance.
+
+If you do not want to use the instance scope to authenticate, you can also do
+so using service account credentials.
+For more information about the format of that option, see the optional ``credentials`` key.
```
Comment (davidoliverSP2) [inline docs/flocker-features/gce-configuration.rst]: How would a user see the "credentials" key - can we provide some more info here about how to access this, where its located, or a link?

File: docs/flocker-features/gce-configuration.rst [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+   This requires that you create a separate `OAuth Service Account <https://developers.google.com/identity/protocols/OAuth2ServiceAccount>`_.
+   When you create a service account in the GCE console you are prompted to download a JSON blob with your credentials. This option should have precisely the content of that JSON blob.
+   For convenience, since JSON is valid YAML, you can just copy the contents of the file directly into your yaml configuration.
+
+The Google GCE / PD driver maintained by ClusterHQ provides :ref:`storage-profiles`.
+The three available profiles are:
+
+* **Gold**: SSD Persistent Disk
+* **Silver**: SSD Persistent Disk
```
Comment (davidoliverSP2) [inline docs/flocker-features/gce-configuration.rst]: Gold and Silver have the same definition here - can we be more specific about what these profiles provide. If they are legitimately the same, should we call that out?

**Comment by davidoliverSP2** [comment]
Thanks @sarum90 - this is really good, thanks so much for taking this on. I've made some amendments to get this building cleanly, and made some suggestions. Happy to discuss this if you think there is more work required for this document above what is suggested.

File: docs/flocker-features/gce-configuration.rst (hunk 3 above)
Comment (justnoise) [inline docs/flocker-features/gce-configuration.rst]: Yep, there are only 2 types of disks we can use on GCE: Standard and SSD.  Pointing out that they're the same would probably be a good idea.

File: docs/flocker-features/gce-configuration.rst [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+.. option:: credentials
+
+   If the credentials parameter is specified, then the GCE driver will use the given service account credentials rather than the instance's compute engine credentials and scope to authenticate with GCE.
+
+   This requires that you create a separate `OAuth Service Account <https://developers.google.com/identity/protocols/OAuth2ServiceAccount>`_.
+   When you create a service account in the GCE console you are prompted to download a JSON blob with your credentials.
+   This option should have precisely the content of that JSON blob.
+   For convenience, since JSON is valid YAML, you can copy the contents of the file directly into your YAML configuration.
+
```
Comment (justnoise) [inline docs/flocker-features/gce-configuration.rst]: Should we include an example of the credentials section of the file?

Comment (sarum90) [inline docs/flocker-features/gce-configuration.rst]: Done

File: docs/flocker-features/gce-configuration.rst (hunk 3 above)
Comment (sarum90) [inline docs/flocker-features/gce-configuration.rst]: Forgot to copy this and accidentally wrote something slightly different. Important thing is it is now called out explicitly.

File: docs/flocker-features/gce-configuration.rst (hunk 2 above)
Comment (sarum90) [inline docs/flocker-features/gce-configuration.rst]: It just is supposed to reference later on in this page... not familiar enough with RST to know how to quickly do that...

**Comment by sarum90** [comment]
Closing in favor of #2681

**Pull request closed without merging**